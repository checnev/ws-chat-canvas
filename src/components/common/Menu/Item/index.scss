.menu {
  &__link {
    padding: 8px 12px;
  }
}
