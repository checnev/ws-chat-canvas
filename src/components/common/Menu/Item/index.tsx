import React from 'react';
import cn from 'utils/cn';
import './index.scss';

class MenuItem extends React.Component {

  render() {
    const menu = cn('menu');
    return (
      <li
        {...this.props}
        className={menu('item')}
        role="menuitem"
      />
    );
  }
}

export default MenuItem;
