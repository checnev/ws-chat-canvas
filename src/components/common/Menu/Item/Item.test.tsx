import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Item from 'components/common/Menu/Item';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render MenuItem', () => {
  act(() => {
    render(<Item>Item Text</Item>, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
