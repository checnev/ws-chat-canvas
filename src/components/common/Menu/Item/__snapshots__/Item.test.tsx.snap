// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MenuItem 1`] = `"<li class=\\"menu__item\\" role=\\"menuitem\\">Item Text</li>"`;
